packages/cli/src/controller/codegen-controller.ts

```
@@ -221,16 +221,32 @@ export async function generateAbis(datasources: DatasourceKind[], projectPath: s
   }
 }
 
-export function processAbis(sortedAssets: Map<string, string>, projectPath: string): abiRenderProps[] {
+function loadFromJsonOrYamlWrapper(filePath: string): abiInterface[] | {abi: abiInterface[]} {
+  return loadFromJsonOrYaml(filePath) as abiInterface[] | {abi: abiInterface[]};
+}
+
+export function processAbis(
+  sortedAssets: Map<string, string>,
+  projectPath: string,
+  loadReadAbi: (filePath: string) => abiInterface[] | {abi: abiInterface[]}
+): abiRenderProps[] {
   const renderInterfaceJobs: abiRenderProps[] = [];
   sortedAssets.forEach((value, key) => {
     const renderProps: abiRenderProps = {name: key, events: [], functions: []};
-    const readAbi = loadFromJsonOrYaml(path.join(projectPath, value)) as abiInterface[] | {abi: abiInterface[]};
+    const readAbi = loadReadAbi(path.join(projectPath, value));
     // We need to use for loop instead of map, due to events/function name could be duplicate,
     // because they have different input, and following ether typegen rules, name also changed
     // we need to find duplicates, and update its name rather than just unify them.
 
-    const abiArray = !Array.isArray(readAbi) && readAbi.abi ? readAbi.abi : (readAbi as unknown as abiInterface[]);
+    let abiArray: abiInterface[] = [];
+
+    if (!Array.isArray(readAbi)) {
+      if (readAbi.abi) {
+        abiArray = readAbi.abi;
+      }
```

We need to throw an error here. Also a test would be good